### ISSUE TITLE
for loop inside mm_loss runs only once

### ISSUE THREAD

**Issue by deepjedi** [fearofchou/mmnet#2]
The for loop inside mm_loss "attempts" to aggregate losses from all the temporal segments but the loop gets executed only once.  With this oblivious error, the code still generates 63% mAP, the result that is presented in the paper.  Is this somehow intended but the code was just simply not cleaned up that a for loop is left there without any use?

**Comment by fearofchou** [comment]
The for loop is for multi-scale version (Mmnet-MS). You are right I should remove the for loop in the code of Mmnet. Sorry for the confusion.

**Comment by deepjedi** [comment]
Thank you for the clarification :) I have another problem, which is not really of an issue related to this code but more of an issue on my TensorFlow version of M&mnet. I have really scrutinized both the documentation of PyTorch and TensorFlow, and have implemented M&mnet in TensorFlow. The network trains up to 53% but it plateaus there. Also, the loss from micro supervision becomes too small. With your code, L_micro stands around 0.4 for the training of 100 epochs. L_micro in my training drops to 0.05 in epoch 2. Do you have any experience with such training behavior?

**Comment by fearofchou** [comment]
Accurately, I have no idea. But I think you need to carefully check the variable of att_sc.data in your TensorFlow version of M&mnet.

**Comment by deepjedi** [comment]
Thank you for the suggestion.  Taking your advice, I have printed the attention scores and there is definitely a strange behaviour.  In the first epoch, it looks normal (i.e. similar to the output of the PyTorch code).  From the second epoch and onward, the first class starts getting a lot of attention and all other classes eventually vanish and become very small numbers.  The attention  mechanism is a simple implementation and I don't see any major discrepancy between PyTorch and Tensorflow in this.  My L_macro is fairly stable but could this (the vanishing attention scores) coming from the prior layers, such as residual blocks even?

**Comment by deepjedi** [comment]
I have resolved this issue now.  I have found out that in TensorFlow tf.stop_gradient needs to be applied to the weights when computing weighted losses.
